### PR TITLE
Compatible with PHP8.1

### PR DESCRIPTION
**Pull request by sy-records**
Fix https://github.com/tencentyun/cos-php-sdk-v5/issues/290

```
\vendor\guzzlehttp\guzzle-services\src\ResponseLocation\XmlLocation.php line 147:
strstr(): Passing null to parameter #1 ($haystack) of type string is deprecated
```

### CODE REVIEW

**Comment by Attax** [comment]
I got the same error,ahaha. please merge the fix ASAP! come on!

**Comment by GrahamCampbell** [comment]
Appologies for missing this one. I'm happy to merge this, or a replacement PR, once test coverage has been added by one of you two.

**Comment by sy-records** [comment]
Hi, @GrahamCampbell Existing tests can cover the modifications. `testVisitsAdditionalProperties`, `testAddsEmptyArraysWhenValueIsMissing`

**Comment by sy-records** [comment]
ping @GrahamCampbell

**Comment by Attax** [comment]
@GrahamCampbell @sy-records said the existed test unit has cover the modifications. Maybe you can merge the PR when your are free. Thank you.

**Comment by GrahamCampbell** [comment]
The existing tests already pass though, indicating that this change doesn't do anything?

**Comment by sy-records** [comment]
Previously the test would produce a waring.

**Comment by sy-records** [comment]
@GrahamCampbell please see https://github.com/lufei/guzzle-services/actions/runs/4729728867/jobs/8392563597

I open `error_reporting=E_ALL`.

File: src/ResponseLocation/XmlLocation.php [hunk 1]
```diff
@@ -80,7 +80,7 @@ public function visit(
     ) {
         $sentAs = $param->getWireName();
         $ns = null;
-        if (strstr($sentAs, ':')) {
+        if (strstr($sentAs ?: '', ':')) {
```
Comment (GrahamCampbell) [inline src/ResponseLocation/XmlLocation.php:83]: `?:` is a bit of a hack. Can we do an explicit null check please?

```
if (null !== $sentAs && strstr($sentAs, ':')) {
```

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/ResponseLocation/XmlLocation.php (hunk 1 above)
Comment (sy-records) [reply src/ResponseLocation/XmlLocation.php]: done.

**Review by GrahamCampbell — APPROVED** [review]

**Comment by GrahamCampbell** [comment]
Excellent, thank you.

**Pull request merged**